core-library/src/main/java/org/silverpeas/core/index/search/model/IndexSearcher.java

```diff
@@ -189,54 +185,62 @@ public MatchingIndexEntry search(String component, String objectId, String objec
-        if (!StringUtil.isDefined(query.getQuery()) && (query.isSearchBySpace() || query.
-            isSearchByComponentType()) && !query.isPeriodDefined()) {
-          rangeQuery = new TermRangeQuery(IndexManager.CREATIONDATE, "1900/01/01", "2200/01/01",
-              true, true);
-        }
-        if (rangeQuery != null) {
-          rangeClauses.add(rangeQuery, BooleanClause.Occur.MUST);
-        }
-        TermRangeQuery rangeQueryOnLastUpdateDate = getRangeQueryOnLastUpdateDate(query);
-        if (rangeQueryOnLastUpdateDate != null) {
-          rangeClauses.add(rangeQueryOnLastUpdateDate, BooleanClause.Occur.MUST);
-        }
-        TermQuery termQueryOnAuthor = getTermQueryOnAuthor(query);
-        if (termQueryOnAuthor != null) {
-          booleanQuery.add(termQueryOnAuthor, BooleanClause.Occur.MUST);
-        }
-        PrefixQuery termQueryOnFolder = getTermQueryOnFolder(query);
-        if (termQueryOnFolder != null) {
-          booleanQuery.add(termQueryOnFolder, BooleanClause.Occur.MUST);
-        }
+        if (query.getMultiFieldQuery() != null) {
+          booleanQueryBuilder.add(getMultiFieldQuery(query), BooleanClause.Occur.MUST);
+        } else {
+          TermRangeQuery rangeQuery = getRangeQueryOnCreationDate(query);
+          if (!StringUtil.isDefined(query.getQuery()) && (query.isSearchBySpace() || query.
+              isSearchByComponentType()) && !query.isPeriodDefined()) {
+            rangeQuery = TermRangeQuery.newStringRange(IndexManager.CREATIONDATE, "1900/01/01", "2200/01/01",
+                true, true);
+          }
+          if (rangeQuery != null) {
+            rangeClausesBuilder.add(rangeQuery, BooleanClause.Occur.MUST);
+          }
+          TermRangeQuery rangeQueryOnLastUpdateDate = getRangeQueryOnLastUpdateDate(query);
+          if (rangeQueryOnLastUpdateDate != null) {
+            rangeClausesBuilder.add(rangeQueryOnLastUpdateDate, BooleanClause.Occur.MUST);
+          }
+          TermQuery termQueryOnAuthor = getTermQueryOnAuthor(query);
+          if (termQueryOnAuthor != null) {
+            booleanQueryBuilder.add(termQueryOnAuthor, BooleanClause.Occur.MUST);
+          }
+          PrefixQuery termQueryOnFolder = getTermQueryOnFolder(query);
+          if (termQueryOnFolder != null) {
+            booleanQueryBuilder.add(termQueryOnFolder, BooleanClause.Occur.MUST);
+          }
 
-        try {
-          Query plainTextQuery = getPlainTextQuery(query, IndexManager.CONTENT);
-          if (plainTextQuery != null) {
-            booleanQuery.add(plainTextQuery, BooleanClause.Occur.MUST);
+          try {
+            Query plainTextQuery = getPlainTextQuery(query, IndexManager.CONTENT);
+            if (plainTextQuery != null) {
+              booleanQueryBuilder.add(plainTextQuery, BooleanClause.Occur.MUST);
+            }
+          } catch (ParseException e) {
+            throw new org.silverpeas.core.index.search.model.ParseException("IndexSearcher", e);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "IndexSearcher" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)